### PR TITLE
[#3] [Chore] Set up the CI with Fastlane and GitHub Actions

### PR DESCRIPTION
**Pull request by kaungkhantsoe**
- #3 

## What happened 👀

Add CI for staging environment.

## Insight 📝

- Removed workflow from original flutter template and reuse it in new workflow file
- Upgrade the Flutter version
- Remove push trigger
- Read `.env.staging` from secrets

## Proof Of Work 📹

You can see the result at https://github.com/nimblehq/flutter-ic-kaung-thieu/actions/runs/4571896695/jobs/8070610425?pr=38

### CODE REVIEW

File: .github/workflows/run_ci_staging.yml [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
-      - 'release/**'
 jobs:
   lint_and_test:
     name: Static code analyze & Unit test
     runs-on: ubuntu-latest
     timeout-minutes: 30
     environment: staging
     steps:
-      - uses: actions/checkout@v2.3.2
+      - name: Checkout Repo
```
Comment (manh-t) [inline .github/workflows/run_ci_staging.yml:14]:
```suggestion
      - name: Checkout repo
```

**Review by manh-t — COMMENTED** [review]
@kaungkhantsoe we should use `dart format` command since `flutter format` is deprecated
image

File: .github/workflows/run_ci_staging.yml [hunk 2]
```diff
@@ -1,27 +1,29 @@
-name: CI
+name: Run CI
 on:
   # Trigger the workflow on push or pull request,
   # but push action is only for the feature branch
   pull_request:
     types: [ opened, synchronize, edited, reopened ]
```
Comment (Thieurom) [inline .github/workflows/run_ci_staging.yml:6]: To cancel in-progress runs of this workflow, you might want to add the following right after this:
```yml
concurrency:
  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
  cancel-in-progress: true
```

Comment (kaungkhantsoe) [reply .github/workflows/run_ci_staging.yml:4]: Fixed in https://github.com/nimblehq/flutter-ic-kaung-thieu/pull/38/commits/31f786889ec11ccfc827e57c441f7df163aeb4ff

**Comment by Thieurom** [comment]
@kaungkhantsoe @manh-t Is it safe to remove the `staging` Environment?

Screenshot 2023-03-31 at 15 05 08

**Review by Thieurom — APPROVED** [review]

**Comment by kaungkhantsoe** [comment]
I am not using `environment` for now. Do you have any plans to use it later @Thieurom ?

**Comment by Thieurom** [comment]
Oops, it seems I accidentally created it! What's wrong with me 😅

**Review by manh-t — APPROVED** [review]

**Pull request merged**